### PR TITLE
Add ReadData for JsonExtensibleData and add ut for MREF project

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiReferenceBuildOutput.cs [hunk 1]
```diff
@@ -303,5 +298,63 @@ private static string GetCompositeName(SpecViewModel svm)
             // If href exists, return name with href
             return ApiBuildOutputUtility.GetXref(svm.Uid, svm.Name, svm.FullName);
         }
+
+        [Serializable]
+        public sealed class JsonExtensionDictionary : VirtualDictionary<string, object>
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiReferenceBuildOutput.cs]: >JsonExtensionDictionary [](start = 28, length = 23)

what about builder pattern?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
>         [ExtensibleMember("modifiers.")]

const ModifiersPrefix

---
Refers to: src/Microsoft.DocAsCode.Build.ManagedReference/BuildOutputs/ApiReferenceBuildOutput.cs:139 in c972119. [](commit_id = c972119f4659d60a615d52b28d91110a9df28d8a, deletion_comment = False)

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**